### DISCUSSION TITLE
Media Element cache mechanism

### DISCUSSION THREAD

**Discussion by EdwardMcFlurry** [CommunityToolkit/Maui#980] [Q&A]
Hello

Is there any/built in/default cache system for Media Element for Maui when playing the Videos over URI. I’m asking this because I’m wondering if can one control how much of the data should one collect/ask for when the state of the media is buffering.

Is there also a way to have few bits/chunks of data already loaded on the player while video is still buffering just for the user to have video loaded fast.

**Comment by EdwardMcFlurry** [comment]
@jfversluis can one override the bits/chunks of few data when playing a video using Uri, is there a default settings for Exo player for Android that I should look into perhaps?

because it takes a while when the current state is in Buffering or/and Opening.

**Reply by jfversluis** [reply]
If you implement a custom handler/mapper mechanism for it I think you should be able to reach the properties/functionalities that you are after for now until we decide to surface something to the .NET MAUI layer.

**Reply by EdwardMcFlurry** [reply]
@jfversluis I’ve reading about Chunk/MediaChunk for Exo player and something like this “DefaultBandwidthMeter” as well but I really have no clue where to begin.

The last missing piece on my App is far as the Media Element goes is the ability to load few chunks of data and starts to play the Video after just few seconds of Buffering on Android 😒 as iOS is ridiculously fast 😀 Was anything like this ever implemented or surfaced in Xamarin.Forms?

I will appreciate any help that you give me.

**Comment by CliffAgius** [comment]
Closed and answered
